lib/tasks/cleanup_db.rake

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+desc 'Remove dangling host associations'
+task cleanup_db: :environment do
+  start = Time.zone.now
+
+  puts 'Beginning cleanup_db.'
+
+  num_deleted = Account.where.not(
+    account_number: Host.select(:account)
+  ).where.not(
+    id: Profile.where.not(account_id: nil).select(:account_id)
+  ).where.not(
+    id: Policy.select(:account_id)
+  ).delete_all
```

Comment:
This task is tested for all cases, both positive and negative, in its associated test. These conditions must all hold for the account to be removed (no hosts & no profiles & no policies).